test/testprocess.c

```diff
@@ -477,12 +540,71 @@ static int process_testSimpleStdinToStdout(void *arg)
     return TEST_ABORTED;
 }
 
+static int process_testBigSimpleStdinToStdout(void *arg)
+{
+    TestProcessData *data = (TestProcessData *)arg;
+    const char *process_args[] = {
+        data->childprocess_path,
+        "--stdin-to-stdout",
+        NULL,
+    };
+    SDL_Process *process = NULL;
+    char *buffer;
+    SDL_bool result;
+    int exit_code;
+    char *text_in = NULL;
+    size_t text_in_buffer_size = 64 * 1024 + 1 + 1; /* 64kiB is the limit on Linux (+1 for null character) */
+    size_t total_read = 0;
+
+    process = SDL_CreateProcess(process_args, SDL_TRUE);
+    SDLTest_AssertCheck(process != NULL, "SDL_CreateProcess()");
+    if (!process) {
+        goto failed;
+    }
+
+    text_in = SDL_malloc(text_in_buffer_size);
+    SDL_memset(text_in, 'a', text_in_buffer_size);
+    text_in[text_in_buffer_size - 1] = '\0';
+
+    SDLTest_AssertPass("About to write to process");
+    result = SDL_WriteProcess(process, text_in, text_in_buffer_size - 1, SDL_TRUE);
```

Comment:
This failure is expected, and documented:
https://github.com/libsdl-org/SDL/blob/27862907c65453e53828e9d892df81c8ce536ea7/include/SDL3/SDL_process.h#L284-L287
This should happen if you send and receive this amount of data in process_testStdinToStdout()

Reply:
Indeed. I was expecting an extra argument to return the actual number of bytes written.

I still think this is a bad designed function, because it is very easy to overload the stdin pipe if the child process is doing something else before reading stdin again.

Reply:
Suggestions are welcome...

Reply:
Okay, we can change it to return the number of bytes written.

Reply:
Actually, I just removed this. Problem solved! :)